pkg/api/v1/validation/validation.go

```diff
@@ -70,12 +70,16 @@ func ValidateResourceRequirements(requirements *v1.ResourceRequirements, fldPath
 
 func validateContainerResourceName(value string, fldPath *field.Path) field.ErrorList {
 	allErrs := validateResourceName(value, fldPath)
-	if len(strings.Split(value, "/")) == 1 {
-		if !helper.IsStandardContainerResourceName(value) {
-			return append(allErrs, field.Invalid(fldPath, value, "must be a standard resource for containers"))
-		}
+
+	// Container resources must be one of:
+	// - A known resource in the `*kubernetes.io/` namespace.
+	// - An extended resource (not in the `*kubernetes.io/` namespace.
+	// - An opaque integer resource [DEPRECATED] (TODO: Remove following deprecation)
+	if !helper.IsStandardContainerResourceName(value) &&
```

Comment:
We unfortunately can't restrict validation without breaking api compatibility.  This would break anyone using the resource field already.

Reply:
Commented on the issue. https://github.com/kubernetes/kubernetes/issues/50658#issuecomment-326174398